haiku-kata-solutions/src/main/java/org/eclipse/collections/haikukata/TextProcessorJDK.java

```diff
@@ -96,24 +96,34 @@ public CharCountsDuplicatesUnique duplicatesAndUnique()
      */
     public record TopVowelAndConsonant(char vowel, char consonant) {};
 
-    public TopVowelAndConsonant topVowelAndConsonant()
-    {
-        // TODO: Find all of the alphabetic letters from this.getHaikuAsChars(), convert them to lowercase,
-        // TODO: count the Character values in a Map and create a List of top 26 occurrences in
-        // TODO: Map.Entry<Character, Long> instances sorted in reverse order by count
-        // Hint: Look at filter, map, mapToObj, collect, Collectors.groupingBy, Collectors.counting
-        // Hint: Also look at entrySet, stream, sorted, Map.Entry.comparingByValue, Comparator.reverseorder, toList
-        List<Map.Entry<Character, Long>> entries = null;
-
-        // TODO: Find the top vowel
-        // Hint: Use the filter method with findFirst, orElseThrow on Stream with the isVowel method below to find
-        // Hint: the top vowel Character value using getKey on Map.Entry.
-        char topVowel = 'a';
-
-        // TODO: Find the top consonant
-        // Hint: Use the filter method with findFirst, orElseThrow on Stream with the isVowel method below to find
-        // Hint: the top consonant Character value using getKey on Map.Entry.
-        char topConsonant = 'b';
+    public TopVowelAndConsonant topVowelAndConsonant() {
+        // Find all of the alphabetic letters from the haiku, convert them to lowercase,
+        // and count the Character values in a Map.
+        Map<Character, Long> charCounts = this.getHaikuAsChars()
+                .stream()
+                .filter(Character::isAlphabetic)
+                .map(Character::toLowerCase)
+                .collect(Collectors.groupingBy(c -> c, Collectors.counting()));
+
+        // Create a List of the top 26 occurrences in Map.Entry<Character, Long> instances sorted in reverse order by count.
+        List<Map.Entry<Character, Long>> entries = charCounts.entrySet()
+                .stream()
+                .sorted(Map.Entry.comparingByValue(Comparator.reverseOrder()))
+                .toList();
+
+        // Find the top vowel.
+        char topVowel = entries.stream()
+                .filter(this::isVowel)
```

Comment:
You need to implement isVowel too

Reply:
I implemented the function

Reply:
do you have to call .getKey() for isVowel?